### DISCUSSION TITLE
nullable out should be better supported

### DISCUSSION THREAD

**Discussion by jklw10** [dotnet/csharplang#6220] [Language Ideas]
![image](https://user-images.githubusercontent.com/8884828/173932050-db671b0b-88c9-4147-8126-8b2066db1c3f.png)

not setting a value to an out value should be allowed if it is nullable.

is not null should return false when comparing an unassigned value, not error.

![image](https://user-images.githubusercontent.com/8884828/173932625-5aa5797f-8e57-45cf-9168-22846cb1f2fd.png)

out should allow for defaults.

**Comment by HaloFour** [comment]
`out` parameters don't have any value assigned to them by default, `null` or otherwise.  Not assigning a value would result in the parameter retaining whatever value it had before the method was called, which could be garbage.

**Reply by HaloFour** [reply]
> out should allow for defaults.

This I find interesting.  It does at least convey intent and might make it easier to write one-liner expression-bodied methods that have `out` parameters.  But it's also using a syntax intended for optional parameters where it doesn't necessarily make sense for the parameter to be optional, and where the metadata doesn't change the semantics of the method today.  What does the default value mean to the caller, if anything?

**Reply by jklw10** [reply]
``bool tryget(out int value) : default value 10``
``bool tryget(out int value : 10) ``
``bool tryget(out int value ? 10) ``
or similar, or if it's nullable just default to null?

**Reply by svick** [reply]
The "out should allow for defaults" part is a duplicate of https://github.com/dotnet/csharplang/issues/1033, which has been marked as "likely never".
